app/javascript/controllers/calendar_controller.js

```diff
@@ -0,0 +1,28 @@
+import { Controller } from "@hotwired/stimulus"
+
+export default class extends Controller {
+  static targets = ["date"]
+
+  connect() {
+    this.dateTargets.forEach((date) => {
+      date.parentElement.addEventListener("click", this.toggleSelect)
+
+      if (date.hasAttribute("selected")) {
+        date.parentElement.setAttribute("selected", true)
+        date.parentElement.classList.add("bg-primary", "text-light")
+      }
+    })
+  }
+
+  toggleSelect(event) {
+    if (event.target.hasAttribute("selected")) {
```

Comment:
Similarly, this `<input type="hidden">` "selected" attribute seems redundant - couldn't you just check if the element is disabled or not?